### ISSUE TITLE
Ubuntu 22.04 - apt upgrade E: Failed to fetch http://downloads.metasploit.com...

### ISSUE THREAD

**Issue by alexandre-pecorilla** [rapid7/metasploit-framework#18192]
Hi there,
While upgrading my Ubuntu today I noticed an issue with Metasploit:

```
$ sudo apt-upgrade
...
The following packages will be upgraded:
  metasploit-framework
1 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Need to get 320 MB of archives.
After this operation, 13.3 kB of additional disk space will be used.
Do you want to continue? [Y/n] Y
Err:1 http://downloads.metasploit.com/data/releases/metasploit-framework/apt lucid/main amd64 metasploit-framework amd64 6.3.26+20230716102705~1rapid7-1
  404  Not Found [IP: 104.66.251.85 80]
E: Failed to fetch http://downloads.metasploit.com/data/releases/metasploit-framework/apt/pool/main/m/metasploit-framework/metasploit-framework_6.3.26%2b20230716102705%7e1rapid7-1_amd64.deb  404  Not Found [IP: 104.66.251.85 80]
```
This is my sources list for metasploit:
```
$ cat /etc/apt/sources.list.d/metasploit-framework.list
deb http://downloads.metasploit.com/data/releases/metasploit-framework/apt lucid main
```

When noticing the issue I first followed @aryanguenthner [steps described here](https://github.com/rapid7/metasploit-framework/issues/16263#issuecomment-1063687165) because I also had a DEPRECATION issue:

```
$ wget http://apt.metasploit.com/metasploit-framework.gpg.key
$ gpg --no-default-keyring --keyring ./metasploit-framework_keyring.gpg --import metasploit-framework.gpg.key
$ gpg --no-default-keyring --keyring ./metasploit-framework_keyring.gpg --export > ./metasploit-framework.gpg
$ sudo mv ./metasploit-framework.gpg /etc/apt/trusted.gpg.d/
$ sudo apt update
```
My metasploit gpg key is in 
**/etc/apt/trusted.gpg.d/metasploit-framework.gpg** but isn't mentioned in **sources.list.d/metasploit-framework.list** 

This might just be a temporary issue with the servers but wanted to signal it anyway.

**Comment by alexandre-pecorilla** [comment]
I'm still unable to install metasploit

**Comment by jemartin-r7** [comment]
Use VPN.

**Comment by bwatters-r7** [comment]
There seem to be a couple things here, and I am going to do some digging.  I'm looking specifically at the deprecation of `apt-key add` that we use in the installer script to add the key for apt.  I have not had a chance to verify, but a possible solution for us would be here: https://askubuntu.com/questions/1286545/what-commands-exactly-should-replace-the-deprecated-apt-key

**Comment by alexandre-pecorilla** [comment]
It works fine now for me as well. Definitely must have been a server issue but what @bwatters-r7 mentioned regarding the apt-key deprecation is a good thing to do for future updates

**Comment by BeetleChunks** [comment]
I started having the same issue as the OP today and think I have identified part of the problem. The error message I am getting is below.

```
Err:1 http://downloads.metasploit.com/data/releases/metasploit-framework/apt lucid/main amd64 metasploit-framework amd64 6.3.40+20231020102649~1rapid7-1
  404  Not Found [IP: 23.64.152.106 80]
E: Failed to fetch http://downloads.metasploit.com/data/releases/metasploit-framework/apt/pool/main/m/metasploit-framework/metasploit-framework_6.3.40%2b20231020102649%7e1rapid7-1_amd64.deb  404  Not Found [IP: 23.64.152.106 80]
```

Note the referenced version in the link within the error message is `6.3.40`.

When I visit `http://downloads.metasploit.com/data/releases/metasploit-framework/apt` and look under the path `pool/main/m/metasploit-framework` the latest version available is `6.3.39`.

I'm not sure why a later version than what is available is being generated, but that does seem to be the reason for the issue I am currently having.

**Comment by BeetleChunks** [comment]
I started having the same issue as the OP today and think I have identified part of the problem. The error message I am getting is below.

```
Err:1 http://downloads.metasploit.com/data/releases/metasploit-framework/apt lucid/main amd64 metasploit-framework amd64 6.3.40+20231020102649~1rapid7-1
  404  Not Found [IP: 23.64.152.106 80]
E: Failed to fetch http://downloads.metasploit.com/data/releases/metasploit-framework/apt/pool/main/m/metasploit-framework/metasploit-framework_6.3.40%2b20231020102649%7e1rapid7-1_amd64.deb  404  Not Found [IP: 23.64.152.106 80]
```

Note the referenced version in the link within the error message is `6.3.40`.

When I visit `http://downloads.metasploit.com/data/releases/metasploit-framework/apt` and look under the path `pool/main/m/metasploit-framework` the latest version available is `6.3.39`.

I'm not sure why a later version than what is available is being generated, but that does seem to be the reason for the issue I am currently having.
